src/metabase/api/pulse.clj

```diff
@@ -59,10 +63,9 @@
 (defendpoint DELETE "/:id"
   "Delete a `Pulse`."
   [id]
-  (let [pulse  (db/sel :one Pulse :id id)
-        result (db/cascade-delete Pulse :id id)]
-    (events/publish-event :pulse-delete (assoc pulse :actor_id *current-user-id*))
-    result))
+  (let [pulse (Pulse id)]
```

Comment:
this should be a `let-404` so that if the pulse doesn't exist we immediately bail with a 404 response.